asm_processor.py

```diff
@@ -320,9 +320,45 @@ def add_section(self, name, sh_type, sh_flags, sh_link, sh_info, sh_addralign, s
     def drop_irrelevant_sections(self):
         # We can only drop sections at the end, since otherwise section
         # references might be wrong. Luckily, these sections typically are.
-        while self.sections[-1].sh_type in [SHT_MIPS_DEBUG, SHT_MIPS_GPTAB]:
+        while self.sections[-1].sh_type in [SHT_MIPS_GPTAB]:
             self.sections.pop()
 
+    def relocate_mdebug(self, mdebug_section, original_offset):
```

Comment:
since we don't use self it would make sense to move this to be a member function on Section